### PR TITLE
Packaging and doc

### PR DESCRIPTION
**Pull request by peijin94**
Put all functions and modules into a package, (named ovrolwasolar)

After install the package, the functions and pipeline can be used with more flexibility (not necessarily be in the same directory)

More detailed see the added part of README in this commit

### CODE REVIEW

**Comment by binchensun** [comment]
@surajit4444mondal I reviewed the changes @peijin94 made. I think this PR should work and can simplify the workflow of installing the package. But it does change the current structure of the repo. Please take a look and let us know if you have any comments before I merge.

File: ovrolwasolar/solar_pipeline.py [hunk 1]
```diff
@@ -415,7 +415,7 @@ def apply_solutions_and_image(msname, bcal, imagename):
     flagdata(vis=msname, mode='rflag', datacolumn='corrected')
     split(vis=msname, outputvis=msname[:-3] + "_selfcalibrated.ms")
     solar_ms = msname[:-3] + "_selfcalibrated.ms"
-    outms = remove_nonsolar_sources(solar_ms)
+    outms = utils.remove_nonsolar_sources(solar_ms)
```
Comment (surajit4444mondal) [inline ovrolwasolar/solar_pipeline.py:418]: remove_nonsolar_sources is in source_subtraction, not in utils.

File: ovrolwasolar/solar_pipeline.py [hunk 2]
```diff
@@ -425,8 +425,8 @@ def apply_solutions_and_image(msname, bcal, imagename):
     else:
         split(vis = solar_ms, outputvis=solar_ms[:-3]+"_sun_selfcalibrated.ms",datacolumn='data')
     outms = solar_ms[:-3] + "_sun_selfcalibrated.ms"
-    outms = remove_nonsolar_sources(outms, remove_strong_sources_only=False)
+    outms = utils.remove_nonsolar_sources(outms, remove_strong_sources_only=False)
```
Comment (surajit4444mondal) [inline ovrolwasolar/solar_pipeline.py:428]: same comment as earlier.

File: ovrolwasolar/solar_pipeline.py [hunk 3]
```diff
@@ -425,8 +425,8 @@ def apply_solutions_and_image(msname, bcal, imagename):
...
     change_phasecenter(outms)
-    run_wsclean(outms, imagename=imagename, automask_thresh=5, uvrange='0', predict=False, imsize=1024, cell='1arcmin')
-    correct_primary_beam(outms, imagename + "-image.fits")
+    utils.run_wsclean(outms, imagename=imagename, automask_thresh=5, uvrange='0', predict=False, imsize=1024, cell='1arcmin')
```
Comment (surajit4444mondal) [inline ovrolwasolar/solar_pipeline.py:430]: run_wsclean is in deconvolve.

File: ovrolwasolar/solar_pipeline.py [hunk 4]
```diff
@@ -425,8 +425,8 @@ def apply_solutions_and_image(msname, bcal, imagename):
...
+    utils.correct_primary_beam(outms, imagename + "-image.fits")
```
Comment (surajit4444mondal) [inline ovrolwasolar/solar_pipeline.py:431]: correct_primary_beam is inside solar_pipeline.py itself.

**Review by surajit4444mondal — COMMENTED** [review]
I have left some comments in the relevant lines, which should be addressed before merging.

File: ovrolwasolar/solar_pipeline.py (hunk 4 above)
Comment (peijin94) [reply ovrolwasolar/solar_pipeline.py:431]: can't find correct_primary_beam in solar_pipeline but found it in utils

Comment (surajit4444mondal) [reply ovrolwasolar/solar_pipeline.py:431]: Right.. my bad.. sorry..

**Comment by peijin94** [comment]
Thanks @surajit4444mondal for all the suggestions, updated solar_pipeline.py accordingly

**Review by binchensun — APPROVED** [review]

**Pull request merged**